sound/soc/sof/loader.c

```diff
@@ -84,7 +84,7 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 offset)
 		ext_hdr = (struct sof_ipc_ext_data_hdr *)ext_data;
 	}
 out:
-	kfree(ext_data);
+	free_page((unsigned long)ext_data);
```

Comment:
I am having a bit of hearburn with those pointer to unsigned long conversions. I can see the benefit of using dev_get_free_pages in the ipc, here I don't see what this change brings - and I find the code more confusing and wondering if this introduces a restriction on addressing?

Reply:
> I am having a bit of hearburn with those pointer to unsigned long conversions. I can see the benefit of using dev_get_free_pages in the ipc, here I don't see what this change brings - and I find the code more confusing and wondering if this introduces a restriction on addressing?

OK, I will remove it.